src/kakarot/instructions/system_operations.cairo

```diff
@@ -406,81 +418,78 @@ namespace CallHelper {
         let value = with_value * stack_value + (1 - with_value) * ctx.call_context.value;
         let args_offset = 2 ** 128 * popped[2 + with_value].high + popped[2 + with_value].low;
         let args_size = 2 ** 128 * popped[3 + with_value].high + popped[3 + with_value].low;
+        let ret_offset = 2 ** 128 * ret_offset_uint256.high + ret_offset_uint256.low;
+        let ret_size = 2 ** 128 * ret_size_uint256.high + ret_size_uint256.low;
+
+        // 2. Gas
+        let available_gas = ctx.call_context.gas_limit - ctx.gas_used;
```

Comment:
we should group this with the gas operations below for the code to be easier to read